gateway/src/main/scala/filodb/gateway/GatewayServer.scala

```diff
@@ -52,21 +69,29 @@ object GatewayServer extends StrictLogging {
   val numContainersSent = Kamon.counter("num-containers-sent")
   val containersSize = Kamon.histogram("containers-size-bytes")
 
+  // Most options are for generating test data
+  class GatewayOptions(args: Seq[String]) extends ScallopConf(args) {
+    val samplesPerSeries = opt[Int](short = 'n', default = Some(100),
+                                    descr="# of samples per time series")
+    val numSeries = opt[Int](short='p', default = Some(20), descr="# of total time series")
+    val sourceConfigPath = trailArg[String](descr="Path to source config, eg conf/timeseries-dev-source.conf")
+    val genHistData = toggle(noshort=true, descrYes="Generate histogram-schema test data and exit")
+    val genPromData = toggle(noshort=true, descrYes="Generate Prometheus-schema test data and exit")
```

Comment:
I think the options should take some config file such as `--genData some-config-file-with-schema` that has details about the schema of data to send. This could be one or multiple entries (in the future when we have multiple schemas per dataset).

Having a dataset definition config file is also useful for dataset creation purposes. We dont need to remember commands, instead create dataset by pointing to a definition file.

Data generation for two types of data columns (double, hist) would be supported, both of which we have now.

Test data generation based on dataset definition also has other benefits where we can test other schemas easily without much setup.

We may need a quick and dirty partKey generator too.

Reply:
I have thought about configurable schema generation.  That would be a good feature, I'd like to save it for a separate PR though and not blow up the scope of this.